### PR TITLE
gui: Diagnostics refresh

### PR DESCRIPTION
**Pull request by jamescowens**
This PR refreshes the diagnostics page code to utilize the researcher model. It also adds parsing of RAC from the client_state.xml file in the underlying researcher. This is needed because the diagnostics will typically be run before someone has magnitude. Magnitude can take more than 24 hours to go above zero, yet when someone attaches one or more eligible projects, especially where they have been crunching before, they will expect diagnostics to pass.

The HasRAC() method follows the eligibility rules in the researcher class, so it simplifies the diagnostics greatly, eliminating a bunch of semi-duplicated code.

The researcher test harness has also been updated to support the parsed RAC and the associated HasRAC() method.

This should also fix reports of diagnostics failing because of the blank external CPID for first project problem.

### CODE REVIEW

File: src/test/gridcoin/researcher_tests.cpp [hunk 1]
```diff
@@ -532,6 +543,9 @@ BOOST_AUTO_TEST_CASE(it_applies_a_provided_team_whitelist)
         BOOST_FAIL("Project 3 does not exist in the mining project map.");
     }
 
+    // HasRAC should be true.
+    //BOOST_CHECK
+
```
Comment (cyrossignol) [inline src/test/gridcoin/researcher_tests.cpp]: ☝️ 🙂

File: src/gridcoin/researcher.h [hunk 2]
```diff
@@ -89,6 +89,8 @@ struct MiningProject
     Cpid m_cpid;        //!< CPID of the BOINC account for the project.
     std::string m_team; //!< Name of the team joined for the project.
     std::string m_url;  //!< URL of the project website.
+    std::string m_s_rac;//!< RAC string of the project.
```
Comment (cyrossignol) [inline src/gridcoin/researcher.h]: Anything use `m_s_rac` besides the intermediate parse step? It seems as if we don't need to keep this in memory. I suggest wrapping the `ExtractXML()` for `<user_expavg_credit>` with `strtold()` for the argument passed to the constructor to initialize `m_rac` directly. That function will return `0.0` already if the parse fails without the need to handle an exception.

**Review by cyrossignol — COMMENTED** [review]
Looks pretty good.

File: src/gridcoin/researcher.h (hunk 2 above)
Comment (jamescowens) [reply src/gridcoin/researcher.h]: I am fine with that.

File: src/test/gridcoin/researcher_tests.cpp (hunk 1 above)
Comment (jamescowens) [reply src/test/gridcoin/researcher_tests.cpp]: Hehe... yeah... 🙂

**Review by cyrossignol — APPROVED** [review]

**Pull request merged**